File: Admin_p/Admin_p.Web/Modules/electroDB/Products/ProductsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Admin_p.electroDB.Columns;

[ColumnsScript("electroDB.Products")]
[BasedOnRow(typeof(ProductsRow), CheckNames = true)]
public class ProductsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int ProductId { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public int StockQuantity { get; set; }
    public string CategoryName { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public string ImagePath { get; set; }
}

File: Admin_p/Admin_p.Web/Modules/ServerTypes/electroDB/ProductsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProductsRow } from "./ProductsRow";

export interface ProductsColumns {
    ProductId: Column<ProductsRow>;
    Name: Column<ProductsRow>;
    Description: Column<ProductsRow>;
    Price: Column<ProductsRow>;
    StockQuantity: Column<ProductsRow>;
    CategoryName: Column<ProductsRow>;
    CreatedAt: Column<ProductsRow>;
    UpdatedAt: Column<ProductsRow>;
    ImagePath: Column<ProductsRow>;
}

export class ProductsColumns extends ColumnsBase<ProductsRow> {
    static readonly columnsKey = 'electroDB.Products';
    static readonly Fields = fieldsProxy<ProductsColumns>();
}